### DISCUSSION TITLE
How can I reuse the commit message that was rejected by a `commit-msg` hook?

### DISCUSSION THREAD

**Discussion by dannyfreeman** [magit/magit#5156] [Support] [answered]
If I write a commit message and it gets rejected by the `commit-msg` hook, the contents of the rejected commit are still present under the file `$(git rev-parse --git-dir)/COMMIT_EDITMSG`. Sometimes when my commit message is very long I want to reuse that commit message and fit it so it passes the commit-msg hook.

Is there a way in magit for me to try to commit again while reusing that message? From the command line you can do something like this

```
git commit --file "$(git rev-parse --git-dir)/COMMIT_EDITMSG" --edit
```

Is there an easy way I could use magit to edit my commit like that? I do not see a way from the commit transient menu? If not, is there a good point of extension in magit where I could enable this behavior?

**Comment by tarsius** [comment, marked as answer]
When you finish or abort a commit, `git-commit.el` saves it to `log-edit-comment-ring`, and you can bring it back from there using `git-commit-{next,prev}-message` on `M-n`/`M-p`, while in a commit message buffer.

Note that the value of this variable does not persist when you exit Emacs.

**Reply by dannyfreeman** [reply]
I did not know about that feature! It's works even better than what I had in mind. Thank you!!!!!
